decidim-core/app/services/decidim/resource_search.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+module Decidim
+  # This is the base class to be used by other search services.
+  # Searchlight documentation: https://github.com/nathanl/searchlight
+  class ResourceSearch < Searchlight::Search
+    # Initialize the Searchlight::Search base class with the options provided.
+    #
+    # scope - The scope used to create the base query
+    # options - A hash of options to modify the search. These options will be
+    #          converted to methods by SearchLight so they can be used on filter #
+    #          methods. (Default {})
+    def initialize(scope, options = {})
+      super(options)
+      @scope = scope
+    end
+
+    # Creates the SearchLight base query.
+    # Check if the option feature was provided.
+    def base_query
+      # raise order_start_time.inspect
+      raise "Missing feature" unless feature
+
+      @scope
+        .page(options[:page] || 1)
+        .per(options[:per_page] || 12)
+        .where(feature: feature)
+    end
+
+    # Handle the category_id filter
+    def search_category_id
+      query.where(decidim_category_id: category_ids)
```

Comment:
Nice although not sure if we should make the assumption that `category_id` will be available in all resources. Maybe we could separate common searches into `Query` classes if we want to share them amongst resources.

In this scenario, something like:

```ruby
def search_category_ids
  CategoryFilterQuery.new(query, categories).query
end
```

or even, if we want to go for the rectify composition way:

```ruby
def query
  ResourceSearch::BaseQuery.new(@scope)
end

def search_category_ids
  query | CategoryFilterQuery.new(categories)
end
```

I feel like this is a good place to use rectify query composition. I would even consider removing `Searchlight`, as it feels like we have some functionality overlap and we could implement this filter ourselves.

Check out [Rectify query composition's documentation](https://github.com/andypike/rectify#composition).